### PR TITLE
Fixes lasers not properly deleting

### LINKED ISSUE #27667: NuLasers can be thrown by explosions
> Prepare for unforeseen consequences.
@DamianX @PaulRitter 
You can now be massdrived by your own laserbeam if you shoot a fueltank in a vacuum.
image
image
What I expected: Cracked crates
What I got: 
image

[bugfix]

### PR DESCRIPTION
**Pull request by PaulRitter**
fixes #27667
~~0% tested cause i got no idea how to test such an edge case~~

~~honestly i got no idea how my change could have implemented this bug, i genuinly assume this was always happening but now since the ACTUAL beam-obj is staying with the user, it gets noticed.~~

~~since i got no idea what causes this bug this is literally my best solution~~

~~regarding #27743: i have no idea how they keep referencing each other, the reference should get cleared in beams.dm L31-33 & rayCastHit.dm L18. @DamianX~~

Edit: this pr should fix lasers not properly deleting, i did not get around to testing it yet

as for the explosion thingy, it is 100% not the explosion code, since there is literally NO way the laser makes it past [this line](https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/game/objects/explosion.dm#L155)

### CODE REVIEW

File: code/modules/projectiles/projectile/beams.dm [hunk 1]
```diff
@@ -68,6 +68,7 @@ var/list/beam_master = list()
 	flag = "laser"
 	eyeblur = 4
 	fire_sound = 'sound/weapons/Laser.ogg'
+	anchored = TRUE
```
Comment (Exxion) [inline code/modules/projectiles/projectile/beams.dm]: Aren't projectiles normally anchored?

Comment (PaulRitter) [reply code/modules/projectiles/projectile/beams.dm]: ah fuck

should explosions throw anchored obj/ should throw_at still throw it even when its anchored?

Comment (Exxion) [reply code/modules/projectiles/projectile/beams.dm]: Explosions definitely shouldn't throw anchored objects
`throw_at()` is up for debate

**Comment by Exxion** [comment]
Is the changelog still accurate?

**Comment by PaulRitter** [comment]
~~no~~ yes

**Comment by Kurfursten** [comment]
Test it

File: code/modules/projectiles/projectile/beams.dm [hunk 2]
```diff
@@ -95,25 +94,26 @@ var/list/beam_master = list()
 
 	if(!gcDestroyed)
 		past_rays += shot_ray
-	else
-		shot_ray.fired_beam = null // hard-delete prevention
 
-	if(isnull(hits) || hits.len == 0)
+	var/distance = MAX_BEAM_DISTANCE;
```
Comment (ShiftyRail) [inline code/modules/projectiles/projectile/beams.dm]: That `;` isn't needed and breaks style convention
```suggestion
	var/distance = MAX_BEAM_DISTANCE
```

**Comment by DamianX** [comment]
Did you end up testing this?

**Comment by PaulRitter** [comment]
nope, haven't come around to doing it yet

**Comment by PaulRitter** [comment]
feel free to mark this derelict and test it yourself, i wont get around to testing this soon

**Comment by DamianX** [comment]
this is so sad
